weblab/entities/tests/test_views.py

```diff
@@ -96,6 +96,26 @@ def test_create_protocol_requires_permissions(self, logged_in_user, client):
         assert response.status_code == 302
         assert '/login/' in response.url
 
+    def test_error_if_later_version(self, logged_in_user, client, helpers):
+        helpers.add_permission(logged_in_user, 'create_model')
+        model = recipes.model.make(author=logged_in_user)
+        helpers.add_version(model, tag_name='v1')
+
+        response = client.post(
+            '/entities/models/%d/versions/new' % model.pk,
+            data={
+                'parent_hexsha': '',
+                'filename[]': ['file1.txt'],
+                'delete_filename[]': [],
+                'mainEntry': ['file1.txt'],
+                'commit_message': 'files',
+                'tag': 'v1',
+                'visibility': 'public',
+            },
+        )
+
+        assert response.status_code == 200
```

Comment:
Looks good. It would be nice to check that we have an expected error message too with this kind of thing.

Reply:
I have done this using `assert "Someone has saved a newer version since you started " in response.rendered_content`. Not sure if there is a better way? Also do I need to check anything else after the post please?

Reply:
That does work. I thought it might be the best option, then I realised that `FormMixin` provides `form` as a variable in the template context for us, so we can use that to test errors more programmatically. Something like
```
assert None in response.context['form'].errors
assert 'Someone has saved a newer version' in response.context['form'].errors[None]
```
I think? I don't think you need to test anything beyond that the error exists, although if you wanted to be paranoid I guess you could check that the latest commit in the repo hasn't changed?

Reply:
Thanks Jonathan, I couldn't get your two lines of code to run. Do I need to specify FormMixin on the test method?

Reply:
You shouldn't do, because it's included in `EntityNewVersionView` which this is testing. Maybe print `response.context` to see what is there?

Reply:
Thanks. I have done a search on response.context and can't see anywhere where the error messages appear.

Reply:
Does the form appear at all?

Reply:
I don't believe so

Reply:
Oh well! Let's stick with your existing approach then :)